app/src/main/java/com/github/geohunt/app/model/database/NetworkMonitor.kt

```diff
@@ -0,0 +1,26 @@
+package com.github.geohunt.app.model.database
+
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ktx.snapshots
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+
+/**
+ * This class provides utility methods for detecting connection state to Firebase's database,
+ * which can be listened to detect disconnection/reconnection events.
+ */
+class NetworkMonitor(
```

Comment:
The class name NetworkMonitor is not very descriptive, consider renaming the class to FirebaseDatabaseConnectionMonitor

Reply:
I would agree with you, but I do not think that we intend to have more network monitors in the future, which would justify creating a `NetworkMonitor` interface and a `FirebaseNetworkMonitor` implementation. We can migrate as you said to an implementation-specific name if we get other implementations.